Add information about building with docker (#63201)
openshift/openshift-docs: contributing_to_docs/tools_and_setup.adoc

* An editor that can strip trailing whitespace, such as
link:https://code.visualstudio.com/[Visual Studio Code].

> === Build with podman/docker

Reviewer: What about 
```
=== Building using the asciibinder container image
```

> === Build with podman/docker

Author: ```suggestion
=== Building using the asciibinder container image
```

> . Install the _docker_ or _link:https://podman.io/docs/installation[podman]_ package.

Author: ```suggestion
. Install the _docker_ or _link:https://podman.io/docs/installation[Podman]_ package.
```

> . Change to the `openshift-docs` directory and build every distribution by entering the `run` command for the tool that you are using. The following example uses Docker and should be identical for Podman on Linux while might require some setup on Mac. 

Reviewer 2: ```suggestion
. Change to the `openshift-docs` directory and build every distribution by entering the `run` command for the tool that you are using. The following example uses Docker and is identical for Podman on Linux. This example might require some setup on a Mac. 
```

> . Change to the `openshift-docs` directory and build every distribution by entering the `run` command for the tool that you are using. The following example uses Docker. If you are using Podman on a Mac, you must perform additional configuration steps provided at the end of this section. 

Author: ```suggestion
. Change to the `openshift-docs` directory and build every distribution by entering the `run` command for the tool that you are using. The following example uses Docker and should be identical for Podman on Linux while might require some setup on Mac. 
```

+
----
$ cd openshift-docs
$ docker run --rm -it -u `id -u` -v `pwd`:/docs:Z quay.io/openshift-cs/asciibinder asciibinder build
----
3. Open the generated HTML file in your web browser. This will be located in the
`openshift-docs/_preview/<distro>/<branch>` directory, with the same path and

> filename as the original `.adoc` file you edited, only it will be with the

Reviewer 2: ```suggestion
filename as the original `.adoc` file you edited, but with an
```

`.html` extension.

> 

Reviewer 2: I needed to perform a lot of additional steps to get this working on a Mac with podman. Thankfully @gabriel-rh was able to help me out. Should we add these steps here? Or do you think users would be able to figure it out? (I could not.)

```
=== Using Podman on a Mac

. Make a temporary directory for Ruby in your home directory for when it runs in the container by using the following commands:
+
----
$ cd
$ mkdir ruby-tmp
$ chmod 1777 ruby-tmp
----
. Initialize the Podman machine for your home directory. You must set the security model to none so that symbolic links work.
+
----
$ podman machine init -v /Users/<username>/:/mnt/top:security_model=none
----
. Start the machine by running the following command:
+
----
$ podman machine start
----
. Connect to the machine by using ssh:
+
----
$ podman machine ssh
----
. Change to the home directory and ensure that the `ruby-tmp` directory is sticky by using the following commands:
+
----
$ pwd
$ cd /mnt/top
$ ls -al ruby-tmp/
----
. Change to your home directory and run the following command:
+
----
$ podman run --rm -it -v /mnt/top/ruby-tmp:/tmp -v `pwd`:/docs quay.io/openshift-cs/asciibinder asciibinder build --distro <distribution name><1>
----
<1> Name of the distribution, for example, `openshift-enterprise` or `openshift-acs`
. Exit the vm when the job is completed. The rendered HTML files are located in the `_preview` directory.

```

Author: I'm not using mac so I can't verify it :(

Author: I asked @mtodor to check it and it looks like docker is working fine without any magic. How about removing `Podman` and having it only for docker? I thought that podman is a drop-in replacement but apparently not.

> $ docker run --rm -it -u `id -u` -v `pwd`:/docs:Z quay.io/openshift-cs/asciibinder asciibinder build

Author: ```
-u `id -u`
```
is for fixing issues with generated files permissions

> $ docker run --rm -it -u `id -u` -v `pwd`:/docs:Z quay.io/openshift-cs/asciibinder asciibinder build

Reviewer: And is there a problem that we see without this in place? I've not noticed anything and have been running without it extensively.

> $ docker run --rm -it -u `id -u` -v `pwd`:/docs:Z quay.io/openshift-cs/asciibinder asciibinder build

Author: ```suggestion
$ docker run --rm -it -v `pwd`:/docs:Z quay.io/openshift-cs/asciibinder asciibinder build
```
